library_generation/model/gapic_config.py

```diff
@@ -22,3 +26,44 @@ class GapicConfig:
 
     def __init__(self, proto_path: str):
         self.proto_path = proto_path
+
+    def __eq__(self, other) -> bool:
+        if not isinstance(other, GapicConfig):
+            return False
+        return self.proto_path == other.proto_path
+
+    def __lt__(self, other) -> bool:
+        self_version = find_version_in_proto_path(self.proto_path)
+        other_version = find_version_in_proto_path(other.proto_path)
+        # if only one config has a version in the proto_path, it is smaller
+        # than the other one.
+        if self_version and (not other_version):
+            return True
+        if (not self_version) and other_version:
+            return False
+
+        self_dirs = self.proto_path.split("/")
+        other_dirs = other.proto_path.split("/")
+        # if both of the configs have no version in the proto_path, the one
+        # with lower depth is smaller.
+        if (not self_version) and (not other_version):
+            return len(self_dirs) < len(other_dirs)
+
+        self_stable = GapicConfig.__is_stable(self_version)
+        other_stable = GapicConfig.__is_stable(other_version)
+        # if only config has a stable version in proto_path, it is smaller than
+        # the other one.
+        if self_stable and (not other_stable):
+            return True
+        if (not self_stable) and other_stable:
+            return False
+        # if two configs have different depth in proto_path, the one with
+        # lower depth is smaller.
+        if len(self_dirs) != len(other_dirs):
+            return len(self_dirs) < len(other_dirs)
+        # otherwise, the one with higher version is smaller.
+        return self_version > other_version
```

Comment:
I'm a bit ignorant to string comparison in python. Is `v2` greater than `v1` because "2" has a higher ASCII value than "1"?

Reply:
> Is v2 greater than v1 because "2" has a higher ASCII value than "1"?

Yes.

I released that string comparison has a bug here: `v10` is smaller than `v2`. I'll change the algorithm.

Reply:
Done.